### PR TITLE
test: Add unit tests to detailed view components

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
## Objective
Add unit tests to the components used on the detailed views for each type of integration (PluginDetailedView, IntegrationDetailedView, SentryAppDetailedView).

### CODE REVIEW

File: tests/js/spec/views/organizationIntegrations/pluginDetailedView.spec.js
Comment (scefali) [inline tests/js/spec/views/organizationIntegrations/pluginDetailedView.spec.js]: @NisanthanNanthakumar Is there no way to test that the modal opens?

Comment (NisanthanNanthakumar) [reply tests/js/spec/views/organizationIntegrations/pluginDetailedView.spec.js]: @scefali unfortunately, it's not a function passed into props that I can mock with Jest. Also the component is out of the scope of the `wrapper` dom, so I can't see if it shows up.

File: tests/js/spec/views/organizationIntegrations/sentryAppDetailedView.spec.jsx
Comment (scefali) [inline tests/js/spec/views/organizationIntegrations/sentryAppDetailedView.spec.jsx]: @NisanthanNanthakumar We should have a sentry app test for one that has a redirectUrl and make sure it redirects and another that doesn't. We should also make sure the mock for creating the installation is called as well.

File: tests/js/spec/views/organizationIntegrations/sentryAppDetailedView.spec.jsx
Comment (scefali) [inline tests/js/spec/views/organizationIntegrations/sentryAppDetailedView.spec.jsx]: @NisanthanNanthakumar Can we also get a test on the uninstall buttons?

File: tests/js/spec/views/organizationIntegrations/sentryAppDetailedView.spec.jsx
Comment (scefali) [inline tests/js/spec/views/organizationIntegrations/sentryAppDetailedView.spec.jsx]: @NisanthanNanthakumar I don't think these tests are adding any value since unpublished and published apps are treated exactly the same on the detail view.

File: tests/js/spec/views/organizationIntegrations/pluginDetailedView.spec.js
Comment (scefali) [reply tests/js/spec/views/organizationIntegrations/pluginDetailedView.spec.js]: @NisanthanNanthakumar You can mock things other than props with jest. But you might need to change the import of the modal like such:
```
import * as modal from 'app/actionCreators/modal';

...

modal.openModal(
```
Then you can mock `openModel` on the `modal` object.

File: tests/js/spec/views/organizationIntegrations/sentryAppDetailedView.spec.jsx
Comment (scefali) [inline tests/js/spec/views/organizationIntegrations/sentryAppDetailedView.spec.jsx]: @NisanthanNanthakumar You should check that the installation endpoint was actually called

File: tests/js/spec/views/organizationIntegrations/sentryAppDetailedView.spec.jsx
Comment (scefali) [reply tests/js/spec/views/organizationIntegrations/sentryAppDetailedView.spec.jsx]: @NisanthanNanthakumar I don't see a test for clicking the `UninstallButton` and seeing if the network request to uninstall was called.

File: tests/js/spec/views/organizationIntegrations/pluginDetailedView.spec.js
Comment (scefali) [inline tests/js/spec/views/organizationIntegrations/pluginDetailedView.spec.js]: @NisanthanNanthakumar Can you change `isHidden` to `true`? A PR I am working on will cause this test to fail unless we change `isHidden` in this test: https://github.com/getsentry/sentry/pull/17165

Comment (NisanthanNanthakumar) [reply tests/js/spec/views/organizationIntegrations/pluginDetailedView.spec.js]: @scefali ok can do, but you need to merge your PR first.

Comment (scefali) [reply tests/js/spec/views/organizationIntegrations/pluginDetailedView.spec.js]: @NisanthanNanthakumar Nah, it will be fine to make that change before I merge my PR

Comment (scefali) [reply tests/js/spec/views/organizationIntegrations/pluginDetailedView.spec.js]: Or if this one gets merged first I can just update the test. We just have to make sure we both don't merge without rebasing or Sentry will break lol

Comment (scefali) [reply tests/js/spec/views/organizationIntegrations/pluginDetailedView.spec.js]: @NisanthanNanthakumar probably safer to just get this PR merged first in case I missed a case. I will fix any tests my PR breaks.

**Review by scefali — APPROVED** [review]

**Pull request merged**
